heareval/predictions/task_predictions.py

```diff
@@ -91,22 +97,74 @@ def forward(self, x):
     def training_step(self, batch, batch_idx):
         # training_step defined the train loop.
         # It is independent of forward
-        x, y = batch
+        x, y, filename, timestamp = batch
         y_hat = self.predictor.forward_logit(x)
-        loss = self.predictor.logit_loss(y_hat, y)
+        # Why do we need .float()?
+        # https://discuss.pytorch.org/t/multi-label-binary-classification-result-type-float-cant-be-cast-to-the-desired-output-type-long/117915
+        loss = self.predictor.logit_loss(y_hat, y.float())
         # Logging to TensorBoard by default
         self.log("train_loss", loss)
         return loss
 
     def validation_step(self, batch, batch_idx):
-        x, y = batch
+        x, y, filenames, timestamps = batch
         y_hat = self.predictor.forward_logit(x)
-        loss = self.predictor.logit_loss(y_hat, y)
         y_pr = self.predictor(x)
-        # Logging to TensorBoard by default
-        self.log("val_loss", loss, prog_bar=True)
+        return {
+            "predictions": y_pr,
+            "predictions_logit": y_hat,
+            "targets": y,
+            "filenames": filenames,
+            "timestamps": timestamps,
+        }
+
+    def validation_epoch_end(self, outputs):
```

Comment:
Cool, good idea to add the validation step here. Instead of trying to fit this all into a single predictor class -- let's have a EventPredictionModel and a ScenePredictionModel?

Reply:
Done